python/qpid_dispatch/management/qdrouter.json

```diff
@@ -1126,6 +1126,106 @@
             }
         },
 
+        "router.config.exchange": {
+            "description":"[EXPERIMENTAL] Defines a topic exchange.",
+            "extends": "configurationEntity",
+            "operations": ["CREATE", "DELETE"],
+            "attributes": {
+                "address": {
```

Comment:
@kgiusti , maybe this should be "addr".  That's the established name for addresses in the config.  (I prefer "address", but it's more important to avoid two different names the same thing.)

Reply:
We haven't been very good at being consistent with naming: the router.node itself has an "address" attribute, while the config.autoLink's address is labeled with 'addr'.

There are a few attributes that have 'addr' as a prefix or a suffix to the full name, e.g.:
addrCount, owningAddr, and externalAddr.

So we're already inconsistent to some degree.  I'd favor using 'address' as the attribute name for an entity that has a single address attribute, and use the "<prefix>Addr" for those entities with more than one address attribute.  So my opinion is to keep the 'address' as is.

However, I'm no UI expert by any means.  Hmmm.... Let me see if I can channel the Ghost of Steve Jobs and see what his reaction is....

<Ghost of Steve Jobs>
Don't be consistent - be Brave!!!
Oh, and you're holding that wrong....
</Ghost of Steve Jobs>

jk! ;D

Reply:
I think address on router.node should change to "addr".  It's the only outlier.  The rest of the config is using "addr" consistently.

Side note: "address" for one address and "addr" for multiple addresses is an automatic user gotcha.  There is zero help for the user in those names.

And data point: The doc for "addr" on autoLink says it is just one address.

Reply:
I don't think there really is any consistency - we've had 'router.config.address' and 'router.address' types from the beginning, not 'router.config.addr' etc.  The compound attribute names do use "Addr" as part of the name (while I don't like that either), but when there's a single address attribute it's either "addr" (router.config.autolink) or "address" (router.node).  There are only two cases so consistency hasn't been established.

I just don't like the "addr" contraction in the user-facing API (and this includes <prefix>Addr names).  We use "addr" all over the _code_ as a shortcut for "address", and I think using "addr" for autolink just seemed natural to us programmers.  But for a non-programmer does dropping the "ess" help in any way?

I feel like I'm bikeshedding this.   Are there any other folks that feel strongly either way?  I'll do whatever the consensus is.

Reply:
Ken and I chatted about how to make things more consistent.

In order to avoid two divergent conventions, I've added a request to change from "addr" to "address" to https://issues.apache.org/jira/browse/DISPATCH-918 .

We looked at how the router currently does statistics and just generally audited the names here. We came to the following changes.

Section | Current | Proposed
-- | -- | --
exchange | alternate | alternateAddress
  | deliveriesReceived | receivedCount
  | deliveriesDropped | droppedCount
  | deliveriesForwarded | forwardedCount
  | deliveriesAlternate | divertedCount
binding | exchange | exchangeName
  | nextHop | nextHopAddress
  | phase | nextHopPhase
  | key | bindingKey
  | deliveriesMatched | matchedCount